workspaces/adventure-pack/goodies/typescript/Map.groupBy/index.ts

```diff
@@ -0,0 +1,29 @@
+declare global {
+  interface MapConstructor {
+    groupBy<K, V>(
+      iterable: Iterable<V>,
+      callbackFn: (value: V, index: number) => K,
+    ): Map<K, V[]>;
+  }
+}
+
+Map.groupBy = function <K, V>(
```

Comment:
One more thing, presumably we'd want to use the built-in version when it's available. But we also want to make sure we're testing our own version even when running in a Node that already has `Map.groupBy`. Take a look at some of the tests for `Iterator.prototype` methods, e.g. `filter` or `map` to see what they do to make sure we're testing our own version. (Spoiler alert, they delete the method before importing.)

```suggestion
Map.groupBy ??= function <K, V>(
```

Reply:
Ah, yeah that makes sense. I was definitely testing the built-in before. Luckily all the tests still pass when ensuring it's actually testing my version. 

I have a feeling we're probably not going to be satisfied with the way I've settled on deleting the built-in but if so I'm looking for idiomatic TS suggestions

Reply:
(I think we were probably still testing the code you wrote, since the `Map.groupBy` function was being overwritten, but with the update here it will be more important to `delete`.)